src/_pytest/pathlib.py

```diff
@@ -32,17 +32,44 @@ def ensure_reset_dir(path):
     ensures the given path is an empty directory
     """
     if path.exists():
-        rmtree(path, force=True)
+        rm_rf(path)
     path.mkdir()
 
 
-def rmtree(path, force=False):
-    if force:
-        # NOTE: ignore_errors might leave dead folders around.
-        #       Python needs a rm -rf as a followup.
-        shutil.rmtree(str(path), ignore_errors=True)
-    else:
-        shutil.rmtree(str(path))
+def rm_rf(path):
+    """Remove the path contents recursively, even if some elements
+    are read-only.
+    """
+
+    def chmod_w(p):
+        import stat
+
+        mode = os.stat(str(p)).st_mode
+        os.chmod(str(p), mode | stat.S_IWRITE)
+
+    def force_writable_and_retry(function, p, excinfo):
+        p = Path(p)
+
+        # for files, we need to recursively go upwards
+        # in the directories to ensure they all are also
+        # writable
+        if p.is_file():
+            for parent in p.parents:
+                chmod_w(parent)
+                # stop when we reach the original path passed to rm_rf
+                if parent == path:
+                    break
+
+        chmod_w(p)
+        try:
+            # retry the function that failed
+            function(str(p))
+        except Exception:
```

Comment:
we should trigger a warning, simply to indicate that it couldnt loose the items, that would massively help the users

Reply:
Good idea, done. 👍